modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+    y <- ncdf4::ncdim_def("latitude", "degrees_north", lat)
+    xytdim <- list(x, y, t)
+    FillValue = NA
+  
+  #STEPS: Download all months in startdate to enddate for given variable
+  # Read CSV, copy to array, add to NetCDF file
+
+  #TEMPERATURE NEON.DP1.00002 Air Temp profile or NEON.DP1.00003 Triple-aspirated T (preferred)
+    airTempLoc = grep("DP1.00002",availProducts)
+    airTemp3Loc = grep("DP1.00003",availProducts)
+    if (length(airTempLoc)!=0) { 
+      airTempDates = unlist(availDates[airTempLoc[1]])
+      airTempGoodDates = which((airTempDates >= startMon) & (airTempDates <= endMon))
+      nairTemp = length(airTempGoodDates)
+      } else { nairTemp = 0 }
+    if (length(airTemp3Loc)!=0) { 
+      airTemp3Dates = unlist(availDates[airTemp3Loc[1]])
+      airTemp3GoodDates = which((airTemp3Dates >= startMon) & (airTemp3Dates <= endMon))
+      nairTemp3 = length(airTemp3GoodDates)
+      } else { nairTemp3 = 0 }
+    if ((length(airTempLoc)==0) && (length(airTemp3Loc)==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available") }
+    if ((nairTemp==0) && (nairTemp3==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available in date range ",startMon," ",endMon)
+    }
+  
+  #define NetCDF variable and create NetCDF file
+    airT.var <- ncvar_def(name = "air_temperature", units = "K", dim = xytdim)
+    nc <- ncdf4::nc_create(new.file, vars = airT.var)  #create netCDF file
+    ncdata <- rep(FillValue,length(datetime))
+    if (nairTemp3>nairTemp) {
+      for (mon in airTemp3Dates[airTemp3GoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[airTemp3Loc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
+          csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+          #Retreive time dimension and figure out where in array to put it
+          csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+          arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+          csvVar <- csvData$tempTripleMean
+          csvQF <- csvData$finalQF
+          csvVar[which(csvQF!=1)]=NA  #triple-asp temp always has QF of 1
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"celsius", "K")
+        }
+      }
+    } else {
+      for (mon in airTempDates[airTempGoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[airTempLoc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
```

Comment:
See regex comment above. Also, if you reuse the same regular expression, it may be a good idea to store it as a string so you only have to write it once.